src/test/run-pass/non-terminate/infinite-loop.rs

```diff
@@ -0,0 +1,29 @@
+use std::{process, thread, time};
+
+fn infinite_loop1() -> u8 {
+    loop {}
+}
+
+fn test1() {
+    let x = infinite_loop1();
+    println!("{}", x);
+    process::exit(1);
+}
+
+fn infinite_loop2() -> u8 {
+    let i = 2;
+    while i > 1 {}
+    1
+}
+
+fn test2() {
+    let x = infinite_loop2();
+    println!("{}", x);
+    process::exit(1);
+}
+
+fn main() {
+    thread::spawn(test1);
+    thread::spawn(test2);
+    thread::sleep(time::Duration::from_millis(5000));
```

Comment:
It seems rather long to sleep for 5 seconds in terms of CI budget?

Reply:
@Centril Reduced to 1s.